### ISSUE TITLE
canu failed with 'gatekeeper store exists, but contains no reads'.

### ISSUE THREAD

**Issue by moanamika** [marbl/canu#194]
Dear all

I am trying to run canu for nanopore-raw sequences. I am running the pipeline on a high performance cluster via `qsub`
I get the following output: 

```

-- This is canu parallel iteration #1, out of a maximum of 2 attempts.
--
-- Final error rates before starting pipeline:
--   
--   genomeSize          -- 22000
--   errorRate           -- 0.048
--   
--   corOvlErrorRate     -- 0.144
--   obtOvlErrorRate     -- 0.144
--   utgOvlErrorRate     -- 0.144
--   
--   obtErrorRate        -- 0.144
--   
--   cnsErrorRate        -- 0.144
--
--
-- BEGIN CORRECTION
--
--
-- Corrected reads saved in '/home/my/Directory//Aurelie_BC01.correctedReads.fasta.gz'.
--
--
-- BEGIN TRIMMING
--
----------------------------------------
-- Starting command on Mon Jul 11 06:47:06 2016 with 254240.7 GB free disk space

    /gpfs/Directory/of/the/cluster/canu/1.3/Linux-amd64/bin/gatekeeperCreate \
      -minlength 1000 \
      -o /home/my/Directory//trimming/Aurelie_BC01.gkpStore.BUILDING \
      /home/my/Directory//trimming/Aurelie_BC01.gkpStore.gkp \
     /home/my/Directory//trimming/Aurelie_BC01.gkpStore.BUILDING.err 2>&1

-- Finished on Mon Jul 11 06:47:06 2016 (lickety-split) with 254240.7 GB free disk space
----------------------------------------
================================================================================
Don't panic, but a mostly harmless error occurred and canu failed.

canu failed with 'gatekeeper store exists, but contains no reads'.
```

When looking at Aurelie_BC01.gkpStore.err:

```
Starting file `/home/my/Directory/Aurelie_BC01.gkpStore.gkp
Loading reads from `/home/my/Directory//Aurelie_BC01.correctedReads.fasta.gz`
Processed 0 lines.
Loaded 0 bp from:

Finished with:
0 warnings (bad base or qv, too short, too Long)

Loaded into store:
0 bp
0 reads.

Skipped (too short):
0 bp (-nan%)
0 reads (-nan%).

gatekeeperCreate finished successfully.

```

Interestingly, the script is working when runnning it locally. Thus, no idea about what`s causing this error message - so every suggestion is highly appreciated!

Thank you very much,

Moana

**Comment by skoren** [comment]
It looks like the corrected reads are empty on the cluster run. Since it works on the local machine, there is probably an environment issue causing correction to fail. Can you post the contents of Directory/canu-scripts/canu.\*.out?

**Comment by moanamika** [comment]
Thank you skoren for the fast reply.
The folder /canu-scripts/ is empty and there`s no canu.*.out file in my case.

Thank you, 
Moana

**Comment by skoren** [comment]
Are you manually submitting a Canu job to the grid? You don't need to do that, Canu will recognize your grid and submit itself with appropriate options. What happens if you run the same Canu command with useGrid=1?

**Comment by moanamika** [comment]
So I am using the following script:

```
#!/bin/sh
#$ -M myemail
#$ -l h_cpu=48:00:00
#$ -l h_vmem=15G
#$ -pe smp 4
#$ -cwd
#$ -N Example-Job
#$ -e out/$JOB_NAME-$JOB_ID.err
#$ -o out/$JOB_NAME-$JOB_ID.out

module load Directory/of/the/cluster/canu/1.3 
canu -d /home/my/Directory/ -p Aurelie_BC01 -genomeSize=0.022m -nanopore-raw /home/my/Directory/Aurelie_BC01.fastq stopOnReadQuality=false -contigFilter="2 500 1.0 1.0 2" -corOutCoverage=400 useGrid=1 maxThreads=$NCORES
```
Using - as you suggested - `useGrid=1` I get the following:

```
-- Detected Java(TM) Runtime Environment '1.8.0_66' (from 'java').
-- Detected 16 CPUs and 126 gigabytes of memory.
-- Detected Sun Grid Engine in '/usr/share/gridengine/ubelix'.
-- Detected Grid Engine environment 'smp'.
--
-- WARNING:  Couldn't determine the SGE resource to request memory.
-- WARNING:  Valid choices are (pick one and supply it to canu):
-- WARNING:    gridEngineMemoryOption="-l h_vmem=MEMORY"
-- WARNING:    gridEngineMemoryOption="-l h_vmem_highmem=MEMORY"
-- WARNING:    gridEngineMemoryOption="-l scratch_files=MEMORY"
-- WARNING:    gridEngineMemoryOption="-l scratch_size=MEMORY"
--
================================================================================
Don't panic, but a mostly harmless error occurred and canu failed.

canu failed with 'can't configure for SGE'.
```

**Comment by skoren** [comment]
That message is saying there is no single consumable memory resource on your grid. You need to check your grid environment with ```qconf -sc``` and look for a consumable memory resource. If you don't have one, it would be best to have your grid administrator add it so you can reserve the memory necessary to run Canu jobs. Then you can specify the appropriate flag as suggested in the message with ```gridEngineMemoryOption="-l <your resource name>=MEMORY"```

**Comment by brianwalenz** [comment]
I think what happened is that canu's built in support for grid was confused by it being directly submitted to the grid.  I've seen cases where canu wants to run, say, 23 jobs, but it ends up running the first job over and over.

So, run again with the built in grid support.  You'll probably need to use options
  gridOptions="-l h_cpu=48:00:00 -M email"
  gridEngineMemoryOption="-l h_vmem=MEMORY"
  gridEngineThreadsOption="-pe smp THREADS"

It seems like it discovered gridEngineThreadsOption, so you possibly can leave that out.

**Comment by moanamika** [comment]
All right, thanks a lot for your help @skoren and @brianwalenz

**Comment by brianwalenz** [comment]
Success?

**Comment by moanamika** [comment]
Yes, indeed. Thank you!

**Comment by KevinMcKernan** [comment]
Just installed some Nanopores and am trying to get Canu to run on my Mac Pro. 
Im just trying to run this on the recommended PacBio test data set and getting this error. I suspect its an install issue as its not recognizing gnuplot (which is installed but perhaps not in the right PATH?).
Any hints greatly appreciated.


GenomeVaporizer:bin kevinmckernan$ ./ canu -p ecoli -d ecoli-auto2 genomeSize=4.8m -pacbio-raw p6.25x.fastq gnuplotTested=true
-bash: ./: is a directory
GenomeVaporizer:bin kevinmckernan$ ./canu -p ecoli -d ecoli-auto2 genomeSize=4.8m -pacbio-raw p6.25x.fastq gnuplotTested=true
--   Reason: image not found
-- Detected Java(TM) Runtime Environment '1.8.0_111' (from 'java').
-- Detected 24 CPUs and 128 gigabytes of memory.
-- No grid engine detected, grid disabled.
--
-- Allowed to run   6 jobs concurrently, and use up to   4 compute threads and   16 GB memory for stage 'bogart (unitigger)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'mhap (overlapper)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'mhap (overlapper)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'mhap (overlapper)'.
-- Allowed to run   6 jobs concurrently, and use up to   4 compute threads and    2 GB memory for stage 'read error detection (overlap error adjustment)'.
-- Allowed to run  24 jobs concurrently, and use up to   1 compute thread  and    1 GB memory for stage 'overlap error adjustment'.
-- Allowed to run   6 jobs concurrently, and use up to   4 compute threads and   21 GB memory for stage 'utgcns (consensus)'.
-- Allowed to run  24 jobs concurrently, and use up to   1 compute thread  and    4 GB memory for stage 'overlap store parallel bucketizer'.
-- Allowed to run  24 jobs concurrently, and use up to   1 compute thread  and    8 GB memory for stage 'overlap store parallel sorting'.
-- Allowed to run  24 jobs concurrently, and use up to   1 compute thread  and    5 GB memory for stage 'overlapper'.
-- Allowed to run   3 jobs concurrently, and use up to   8 compute threads and    8 GB memory for stage 'overlapper'.
-- Allowed to run   3 jobs concurrently, and use up to   8 compute threads and    8 GB memory for stage 'overlapper'.
-- Allowed to run   6 jobs concurrently, and use up to   4 compute threads and    8 GB memory for stage 'meryl (k-mer counting)'.
-- Allowed to run  12 jobs concurrently, and use up to   2 compute threads and   10 GB memory for stage 'falcon_sense (read correction)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'minimap (overlapper)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'minimap (overlapper)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'minimap (overlapper)'.
--
-- This is canu parallel iteration #1, out of a maximum of 2 attempts.
--
-- Final error rates before starting pipeline:
--   
--   genomeSize          -- 4800000
--   errorRate           -- 0.015
--   
--   corOvlErrorRate     -- 0.045
--   obtOvlErrorRate     -- 0.045
--   utgOvlErrorRate     -- 0.045
--   
--   obtErrorRate        -- 0.045
--   
--   cnsErrorRate        -- 0.045
--
--
-- BEGIN CORRECTION
--
----------------------------------------
-- Starting command on Tue Dec 27 11:32:05 2016 with 1056.666 GB free disk space

    /Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/gatekeeperCreate \
      -minlength 1000 \
      -o /Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/ecoli-auto2/correction/ecoli.gkpStore.BUILDING \
      /Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/ecoli-auto2/correction/ecoli.gkpStore.gkp \
    > /Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/ecoli-auto2/correction/ecoli.gkpStore.BUILDING.err 2>&1
sh: line 4: 32057 Trace/BPT trap: 5       /Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/gatekeeperCreate -minlength 1000 -o /Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/ecoli-auto2/correction/ecoli.gkpStore.BUILDING /Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/ecoli-auto2/correction/ecoli.gkpStore.gkp > /Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/ecoli-auto2/correction/ecoli.gkpStore.BUILDING.err 2>&1

-- Finished on Tue Dec 27 11:32:05 2016 (lickety-split) with 1056.666 GB free disk space
----------------------------------------
ERROR:
ERROR:  Failed with exit code 133.  (rc=34048)
ERROR:
================================================================================
Don't panic, but a mostly harmless error occurred and canu failed.

canu failed with 'gatekeeper store exists, but contains no reads'.

**Comment by skoren** [comment]
No, the error message seems to indicate there are no sequences loaded in the store. What is the output in: ```/Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/ecoli-auto2/correction/ecoli.gkpStore.BUILDING.err```

**Comment by KevinMcKernan** [comment]
Thank you for such a prompt reply!
Don't seem to have that file.

GenomeVaporizer:correction kevinmckernan$ ls
ecoli.gkpStore.err	ecoli.gkpStore.gkp
GenomeVaporizer:correction kevinmckernan$  pwd
/Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/ecoli-auto2/correction
GenomeVaporizer:correction kevinmckernan$ 

I also installed 1.3 and it gives a different error (DF) that was listed as addressed in 1.4. Don't know if that helps or confuses things?

enomeVaporizer:bin kevinmckernan$ ./canu -p ecoli -d ecoli-auto genomeSize=4.8m -pacbio-raw p6.25x.fastq
dyld: lazy symbol binding failed: Symbol not found: _Perl_xs_handshake
  Referenced from: /Users/kevinmckernan/canu-1.3/Darwin-amd64/bin/lib/canu/lib/perl5/darwin-thread-multi-2level/auto/Filesys/Df/Df.bundle
  Expected in: flat namespace

dyld: Symbol not found: _Perl_xs_handshake
  Referenced from: /Users/kevinmckernan/canu-1.3/Darwin-amd64/bin/lib/canu/lib/perl5/darwin-thread-multi-2level/auto/Filesys/Df/Df.bundle
  Expected in: flat namespace

Trace/BPT trap: 5
GenomeVaporizer:bin kevinmckernan$ perl canu -p ecoli -d ecoli-auto genomeSize=4.8m -pacbio-raw p6.25x.fastq
dyld: lazy symbol binding failed: Symbol not found: _Perl_xs_handshake
  Referenced from: /Users/kevinmckernan/canu-1.3/Darwin-amd64/bin/lib/canu/lib/perl5/darwin-thread-multi-2level/auto/Filesys/Df/Df.bundle
  Expected in: flat namespace

dyld: Symbol not found: _Perl_xs_handshake
  Referenced from: /Users/kevinmckernan/canu-1.3/Darwin-amd64/bin/lib/canu/lib/perl5/darwin-thread-multi-2level/auto/Filesys/Df/Df.bundle
  Expected in: flat namespace

Trace/BPT trap: 5

**Comment by skoren** [comment]
What's the contents of the ecoli.gkpStore.err file? Also, what happens if run the binary gatekeeperCreate from the canu bin path?

**Comment by KevinMcKernan** [comment]
GenomeVaporizer:correction kevinmckernan$ more ecoli.gkpStore.gkp 
########################################
#  pacbio-raw: /Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/p6.25x.fastq
#
name   p6.25x
preset pacbio-raw
/Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/p6.25x.fastq

GenomeVaporizer:correction kevinmckernan$ more ecoli.gkpStore.err 
dyld: Library not loaded: /opt/local/lib/libgcc/libgomp.1.dylib
  Referenced from: /Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/gatekeeperCreate
  Reason: image not found

do I need dyld library?

**Comment by KevinMcKernan** [comment]
GenomeVaporizer:bin kevinmckernan$ ./gatekeeperCreate 
dyld: Library not loaded: /opt/local/lib/libgcc/libgomp.1.dylib
  Referenced from: /Users/kevinmckernan/Downloads/canu-1.4/Darwin-amd64/bin/./gatekeeperCreate
  Reason: image not found
Trace/BPT trap: 5

**Comment by skoren** [comment]
The problem is the binaries are built with support for OpenMP which Canu uses for multi-threading but your system doesn't have OpenMP (because apple removed it from their default compiler a while back). You can either compile Canu from source in which case it will disable OpenMP and you will lose most parallelism or you can install a gnu compiler or an updated clang instead of apples default clang compiler.

**Comment by KevinMcKernan** [comment]
I think I have gcc installed. I'll give that a go on 1.4 and see where that gets me... thanks

**Comment by KevinMcKernan** [comment]
I downloaded clang.

The instructions on for installation are to use make but the canu-1.3/src directory only has /utgcns


gzip -dc canu-1.3.tar.gz | tar -xf -
cd canu-1.3/src
make -j8
cd ..

where do I point clang?

**Comment by skoren** [comment]
You should use the 1.4 release and the src directory has more than just utgcns:
```
% wget https://github.com/marbl/canu/archive/v1.4.tar.gz
% tar xvf v1.4.tar.gz 
% cd canu-1.4src/
% ls
AS_UTL			bogart-analysis		falcon_sense		meryl			overlapInCore
AS_global.C		bogus			fastq-utilities		mhap			overlapInCore-analysis
AS_global.H		canu_version_update.pl	main.mk			minimap			pipelines
Makefile		correction		merTrim			overlapBasedTrimming	stores
bogart			erateEstimate		mercy			overlapErrorAdjustment	utgcns
```
If you have clang installed and set as default (you can check with `which gcc`` and ```gcc --version```) then you can just run make and it will use the appropriate compiler.

**Comment by KevinMcKernan** [comment]
Thank you! Looks like I have to remove the old clang.

I think Make is still pointing the default in /user/bin/

/fastq-utilities/fastqSample.o -lcanu
c++ -o /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/fastqSimulate -pthread -lm -D_GLIBCXX_PARALLEL -L/Users/kevinmckernan/canu-1.4/Darwin-amd64/bin /Users/kevinmckernan/canu-1.4/Darwin-amd64/obj/fastqSimulate/fastq-utilities/fastqSimulate.o -lcanu
c++ -o /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/fastqSimulate-sort -pthread -lm -D_GLIBCXX_PARALLEL -L/Users/kevinmckernan/canu-1.4/Darwin-amd64/bin /Users/kevinmckernan/canu-1.4/Darwin-amd64/obj/fastqSimulate-sort/fastq-utilities/fastqSimulate-sort.o -lcanu
clang: warning: argument unused during compilation: '-pthread'
clang: warning: argument unused during compilation: '-pthread'
clang: warning: argument unused during compilation: '-pthread'
clang: warning: argument unused during compilation: '-pthread'
clang: warning: argument unused during compilation: '-pthread'

Success!
canu installed in /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/canu

**Comment by KevinMcKernan** [comment]
I think I sorted that out. Thank you for all of the help. Next detail is gnuplot path problems..

GenomeVaporizer:bin kevinmckernan$ ./canu  -p ecoli -d ecoli-auto  genomeSize=4.8m  -pacbio-raw p6.25x.fastq gnuplotTested=true
-- Canu v0.0 (+0 commits) r0 unknown-hash-tag-no-repository-available.
-- Detected Java(TM) Runtime Environment '1.8.0_111' (from 'java').
-- Detected 24 CPUs and 128 gigabytes of memory.
-- No grid engine detected, grid disabled.
--
-- Allowed to run   6 jobs concurrently, and use up to   4 compute threads and   16 GB memory for stage 'bogart (unitigger)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'mhap (overlapper)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'mhap (overlapper)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'mhap (overlapper)'.
-- Allowed to run   6 jobs concurrently, and use up to   4 compute threads and    2 GB memory for stage 'read error detection (overlap error adjustment)'.
-- Allowed to run  24 jobs concurrently, and use up to   1 compute thread  and    1 GB memory for stage 'overlap error adjustment'.
-- Allowed to run   6 jobs concurrently, and use up to   4 compute threads and   21 GB memory for stage 'utgcns (consensus)'.
-- Allowed to run  24 jobs concurrently, and use up to   1 compute thread  and    4 GB memory for stage 'overlap store parallel bucketizer'.
-- Allowed to run  24 jobs concurrently, and use up to   1 compute thread  and    8 GB memory for stage 'overlap store parallel sorting'.
-- Allowed to run  24 jobs concurrently, and use up to   1 compute thread  and    5 GB memory for stage 'overlapper'.
-- Allowed to run   3 jobs concurrently, and use up to   8 compute threads and    8 GB memory for stage 'overlapper'.
-- Allowed to run   3 jobs concurrently, and use up to   8 compute threads and    8 GB memory for stage 'overlapper'.
-- Allowed to run   6 jobs concurrently, and use up to   4 compute threads and    8 GB memory for stage 'meryl (k-mer counting)'.
-- Allowed to run  12 jobs concurrently, and use up to   2 compute threads and   10 GB memory for stage 'falcon_sense (read correction)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'minimap (overlapper)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'minimap (overlapper)'.
-- Allowed to run   2 jobs concurrently, and use up to  12 compute threads and    6 GB memory for stage 'minimap (overlapper)'.
--
-- This is canu parallel iteration #1, out of a maximum of 2 attempts.
--
-- Final error rates before starting pipeline:
--   
--   genomeSize          -- 4800000
--   errorRate           -- 0.015
--   
--   corOvlErrorRate     -- 0.045
--   obtOvlErrorRate     -- 0.045
--   utgOvlErrorRate     -- 0.045
--   
--   obtErrorRate        -- 0.045
--   
--   cnsErrorRate        -- 0.045
--
--
-- BEGIN CORRECTION
--
----------------------------------------
-- Starting command on Tue Dec 27 14:19:09 2016 with 1055.408 GB free disk space

    /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/gatekeeperCreate \
      -minlength 1000 \
      -o /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.gkpStore.BUILDING \
      /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.gkpStore.gkp \
    > /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.gkpStore.BUILDING.err 2>&1

-- Finished on Tue Dec 27 14:19:11 2016 (2 seconds) with 1055.378 GB free disk space
----------------------------------------
--
-- WARNING: gnuplot failed; no plots will appear in HTML output.
--
----------------------------------------
--
-- In gatekeeper store '/Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.gkpStore':
--   Found 12528 reads.
--   Found 115899341 bases (24.14 times coverage).
--
--   Read length histogram (one '*' equals 20.62 reads):
--        0    999      0 
--     1000   1999   1444 **********************************************************************
--     2000   2999   1328 ****************************************************************
--     3000   3999   1065 ***************************************************
--     4000   4999    774 *************************************
--     5000   5999    668 ********************************
--     6000   6999    619 ******************************
--     7000   7999    618 *****************************
--     8000   8999    607 *****************************
--     9000   9999    560 ***************************
--    10000  10999    523 *************************
--    11000  11999    478 ***********************
--    12000  12999    429 ********************
--    13000  13999    379 ******************
--    14000  14999    366 *****************
--    15000  15999    353 *****************
--    16000  16999    329 ***************
--    17000  17999    297 **************
--    18000  18999    294 **************
--    19000  19999    283 *************
--    20000  20999    251 ************
--    21000  21999    195 *********
--    22000  22999    152 *******
--    23000  23999    132 ******
--    24000  24999     75 ***
--    25000  25999     66 ***
--    26000  26999     56 **
--    27000  27999     44 **
--    28000  28999     35 *
--    29000  29999     16 
--    30000  30999     21 *
--    31000  31999     18 
--    32000  32999     11 
--    33000  33999      8 
--    34000  34999      6 
--    35000  35999      6 
--    36000  36999     10 
--    37000  37999      2 
--    38000  38999      3 
--    39000  39999      2 
--    40000  40999      2 
--    41000  41999      2 
--    42000  42999      1 
-- Meryl attempt 1 begins.
----------------------------------------
-- Starting concurrent execution on Tue Dec 27 14:19:11 2016 with 1055.378 GB free disk space (1 processes; 6 concurrently)

    /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/0-mercounts/meryl.sh 1 > /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/0-mercounts/meryl.000001.out 2>&1

**Comment by skoren** [comment]
Gnuplot is optional and you can run without it. When you set gnuplotTested=true, it will skip trying to determine what options/version of gnuplot you have and let it fail. If you run without that, it will tell you issues with your gnuplot installation that would need to be fixed to get the plots generated.

**Comment by KevinMcKernan** [comment]
Here is the end of the assembly... failed to create the overlap store.
but otherwise seems to have done something.

-- Finished on Tue Dec 27 14:19:56 2016 (lickety-split) with 1055.029 GB free disk space
----------------------------------------
--
-- WARNING: gnuplot failed; no plots will appear in HTML output.
--
----------------------------------------
--
-- WARNING: gnuplot failed; no plots will appear in HTML output.
--
----------------------------------------
-- For mhap overlapping, set repeat k-mer threshold to 1157.
--
-- Found 115711421 16-mers; 89186744 distinct and 77230204 unique.  Largest count 37954.
--
-- OVERLAPPER (mhap) (correction)
--
-- Set corMhapSensitivity=high based on read coverage of 24.
--
-- PARAMETERS: hashes=768, minMatches=2, threshold=0.73
--
-- Given 6 GB, can fit 9000 reads per block.
-- For 3 blocks, set stride to 2 blocks.
-- Logging partitioning to '/Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/partitioning.log'.
-- Configured 2 mhap precompute jobs.
-- Configured 2 mhap overlap jobs.
-- mhap precompute attempt 1 begins with 0 finished, and 2 to compute.
----------------------------------------
-- Starting concurrent execution on Tue Dec 27 14:19:58 2016 with 1055.378 GB free disk space (2 processes; 2 concurrently)

    /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/precompute.sh 1 > /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/precompute.000001.out 2>&1
    /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/precompute.sh 2 > /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/precompute.000002.out 2>&1

-- Finished on Tue Dec 27 14:26:42 2016 (404 seconds) with 1054.971 GB free disk space
----------------------------------------
-- All 2 mhap precompute jobs finished successfully.
-- mhap attempt 1 begins with 0 finished, and 2 to compute.
----------------------------------------
-- Starting concurrent execution on Tue Dec 27 14:26:42 2016 with 1054.971 GB free disk space (2 processes; 2 concurrently)

    /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/mhap.sh 1 > /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/mhap.000001.out 2>&1
    /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/mhap.sh 2 > /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/mhap.000002.out 2>&1

-- Finished on Tue Dec 27 14:26:56 2016 (14 seconds) with 1054.919 GB free disk space
----------------------------------------
-- Found 2 mhap overlap output files.
----------------------------------------
-- Starting command on Tue Dec 27 14:26:56 2016 with 1054.919 GB free disk space

    /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ovStoreBuild \
     -O /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.ovlStore.BUILDING \
     -G /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.gkpStore \
     -M 2-8 \
     -L /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/ovljob.files \
     > /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.ovlStore.err 2>&1
sh: line 5: 56914 Segmentation fault: 11  /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ovStoreBuild -O /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.ovlStore.BUILDING -G /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.gkpStore -M 2-8 -L /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/ovljob.files > /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.ovlStore.err 2>&1

-- Finished on Tue Dec 27 14:26:56 2016 (lickety-split) with 1054.919 GB free disk space
----------------------------------------
ERROR:
ERROR:  Failed with exit code 139.  (rc=35584)
ERROR:
================================================================================
Don't panic, but a mostly harmless error occurred and canu failed.

Disk space available:  1054.919 GB

Last 50 lines of the relevant log file (/Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.ovlStore.err):

Found 497374 (0.50 million) overlaps.
Configuring for 2.00 GB to 8.00 GB memory and 240 open files.
Will sort using 1 files; 58720256 (58.72 million) overlaps per bucket; 2.00 GB memory per bucket
  bucket   1 has 497374 olaps.
  bucket   2 has 0 olaps.
Will sort 0.497 million overlaps per bucket, using 1 buckets 0.26 GB per bucket.


-- BUCKETIZING --

-  Bucketizing '/Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/results/000001.ovb'

canu failed with 'failed to create the overlap store'.

GenomeVaporizer:bin kevinmckernan$

**Comment by skoren** [comment]
There may still be some issue in your compilation/linking. What is the output in ecoli-auto/correction/1-overlapper/mhap.\*.out and ecoli-auto/correction/1-overlapper/precompute.\*.out? The speed of the mhap jobs in particular seems too fast.

**Comment by KevinMcKernan** [comment]
GenomeVaporizer:1-overlapper kevinmckernan$ more mhap.000001.out 
Running job 1 based on command line options.
Running block 000001 in query 000001
Running with these settings:
--filter-threshold = 5.0E-6
--help = false
--max-shift = 0.2
--min-olap-length = 116
--min-store-length = 0
--no-rc = false
--no-self = false
--no-tf = false
--num-hashes = 768
--num-min-matches = 2
--num-threads = 12
--ordered-kmer-size = 12
--ordered-sketch-size = 1536
--repeat-idf-scale = 10.0
--repeat-weight = 0.9
--settings = 0
--store-full-id = false
--supress-noise = 0
--threshold = 0.73
--version = false
-f = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/0-mercounts/ecoli.ms16.frequentMers.ignore.gz
-h = false
-k = 16
-p = 
-q = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/queries/000001
-s = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks/000001.dat

Reading in filter file /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/0-mercounts/ecoli.ms16.frequentMers.ignore.gz.
Time (s) to read filter file: 0.12833940400000002
Read in k-mer filter for sizes: [16]
Processing files for storage in reverse index...
Current # sequences loaded and processed from file: 5000...
Current # sequences loaded and processed from file: 10000...
Current # sequences loaded and processed from file: 15000...
Current # sequences stored: 5000...
Current # sequences stored: 10000...
Current # sequences stored: 15000...
Stored 18000 sequences in the index.
Processed 18000 unique sequences (fwd and rev).
Time (s) to read and hash from file: 7.290230133000001
Time (s) to score and output to self: 3.8676062360000003
Opened fasta file /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks/000002.dat.
Processed 3528 to sequences.
Time (s) to score, hash to-file, and output: 2.2108502530000003
Total scoring time (s): 6.081159748
Total time (s): 13.371481483
MinHash search time (s): 8.962497284000001
Total matches found: 228171
Average number of matches per lookup: 18.21288314176245
Average number of table elements processed per lookup: 841.0895593869732
Average number of table elements processed per match: 46.18102212814074
Average % of hashed sequences hit per lookup: 0.5838787782034909
Average % of hashed sequences hit that are matches: 17.329399166078062
Average % of hashed sequences fully compared that are matches: 87.42753359414215

**Comment by KevinMcKernan** [comment]
GenomeVaporizer:1-overlapper kevinmckernan$ more mhap.000002.out 
Running job 2 based on command line options.
mkdir: /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/results: File exists
Running block 000002 in query 000002
Running with these settings:
--filter-threshold = 5.0E-6
--help = false
--max-shift = 0.2
--min-olap-length = 116
--min-store-length = 0
--no-rc = false
--no-self = false
--no-tf = false
--num-hashes = 768
--num-min-matches = 2
--num-threads = 12
--ordered-kmer-size = 12
--ordered-sketch-size = 1536
--repeat-idf-scale = 10.0
--repeat-weight = 0.9
--settings = 0
--store-full-id = false
--supress-noise = 0
--threshold = 0.73
--version = false
-f = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/0-mercounts/ecoli.ms16.frequentMers.ignore.gz
-h = false
-k = 16
-p = 
-q = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/queries/000002
-s = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks/000002.dat

Reading in filter file /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/0-mercounts/ecoli.ms16.frequentMers.ignore.gz.
Time (s) to read filter file: 0.128937297
Read in k-mer filter for sizes: [16]
Processing files for storage in reverse index...
Current # sequences loaded and processed from file: 5000...
Current # sequences stored: 5000...
Stored 7056 sequences in the index.
Processed 7056 unique sequences (fwd and rev).
Time (s) to read and hash from file: 1.9602309740000001
Time (s) to score and output to self: 2.049521586
Total scoring time (s): 2.050726988
Total time (s): 4.011133129
MinHash search time (s): 7.730753916
Total matches found: 20516
Average number of matches per lookup: 5.815192743764173
Average number of table elements processed per lookup: 891.6822562358277
Average number of table elements processed per match: 153.3366640670696
Average % of hashed sequences hit per lookup: 0.664617462631311
Average % of hashed sequences hit that are matches: 12.400345730052525
Average % of hashed sequences fully compared that are matches: 85.84818813289814

**Comment by KevinMcKernan** [comment]
GenomeVaporizer:1-overlapper kevinmckernan$ more precompute.*.out
Running job 1 based on command line options.
Dumping reads from 1 to 9000 (inclusive).
Starting mhap precompute.
Running with these settings:
--filter-threshold = 5.0E-6
--help = false
--max-shift = 0.2
--min-olap-length = 116
--min-store-length = 0
--no-rc = false
--no-self = false
--no-tf = false
--num-hashes = 768
--num-min-matches = 2
--num-threads = 12
--ordered-kmer-size = 12
--ordered-sketch-size = 1536
--repeat-idf-scale = 10.0
--repeat-weight = 0.9
--settings = 0
--store-full-id = false
--supress-noise = 0
--threshold = 0.73
--version = false
-f = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/0-mercounts/ecoli.ms16.frequentMers.ignore.gz
-h = false
-k = 16
-p = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks/000001.fasta
-q = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks
-s = 

Reading in filter file /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/0-mercounts/ecoli.ms16.frequentMers.ignore.gz.
Time (s) to read filter file: 0.361922104
Read in k-mer filter for sizes: [16]
Processing FASTA files for binary compression...
Current # sequences loaded and processed from file: 5000...
Current # sequences loaded and processed from file: 10000...
Current # sequences loaded and processed from file: 15000...
Processed 18000 sequences (fwd and rev).
Read, hashed, and stored file /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks/000001.fasta to /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks/000001.dat.
Time (s): 401.17552213100004
Total time (s): 401.175811145
...skipping...
Running job 2 based on command line options.
mkdir: /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks: File exists
Dumping reads from 9001 to 12528 (inclusive).
Starting mhap precompute.
Running with these settings:
--filter-threshold = 5.0E-6
--help = false
--max-shift = 0.2
--min-olap-length = 116
--min-store-length = 0
--no-rc = false
--no-self = false
--no-tf = false
--num-hashes = 768
--num-min-matches = 2
--num-threads = 12
--ordered-kmer-size = 12
--ordered-sketch-size = 1536
--repeat-idf-scale = 10.0
--repeat-weight = 0.9
--settings = 0
--store-full-id = false
--supress-noise = 0
--threshold = 0.73
--version = false
-f = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/0-mercounts/ecoli.ms16.frequentMers.ignore.gz
-h = false
-k = 16
-p = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks/000002.fasta
-q = /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks
-s = 

Reading in filter file /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/0-mercounts/ecoli.ms16.frequentMers.ignore.gz.
Time (s) to read filter file: 0.36071762
Read in k-mer filter for sizes: [16]
Processing FASTA files for binary compression...
Current # sequences loaded and processed from file: 5000...
Processed 7056 sequences (fwd and rev).
Read, hashed, and stored file /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks/000002.fasta to /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/blocks/000002.dat.
Time (s): 186.25847302900002
Total time (s): 186.25873358500002
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~

**Comment by skoren** [comment]
All those outputs look identical to mine and I've confirmed I can run through the example with 1.4 either on Linux or OS X 10.10. Can you try recompiling with debug
```
cd canu-1.4/src
make clean && make BUILDDEBUG=1
```
Then, re-run the failed command
```
/Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ovStoreBuild \
     -O /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.ovlStore.BUILDING \
     -G /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.gkpStore \
     -M 2-8 \
     -L /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/ovljob.files \
```
within GDB and report a stack trace? Can you also share the 1-overlapper/results/\*.ovb files (you can upload them here: ftp://ftp.cbcb.umd.edu/incoming/sergek so I can confirm if they are formatted correctly.

**Comment by KevinMcKernan** [comment]
Not certain I know what GDB is? sorry for my ignorance. I recompiled and ran the command below.

> /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ovStoreBuild -O /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.ovlStore.BUILDING -G /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.gkpStore -M 2-8 -L /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/ovljob.files
Found 994748 (0.99 million) overlaps.
Configuring for 2.00 GB to 8.00 GB memory and 240 open files.
Will sort using 1 files; 58720256 (58.72 million) overlaps per bucket; 2.00 GB memory per bucket
  bucket   1 has 994748 olaps.
  bucket   2 has 0 olaps.
Will sort 0.995 million overlaps per bucket, using 1 buckets 0.28 GB per bucket.


-- BUCKETIZING --

ERROR:  '/Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.ovlStore.BUILDING' is a valid ovStore; cannot create a new one.

**Comment by skoren** [comment]
I'm wondering if your error is related to issue #319. GDB is the debugger for C programs. So if you run the following in the Canu src:
```make clean && make BUILDDEBUG=1```
then the following from the ```/Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/``` folder
```
rm -rf ecoli.ovlStore.BUILDING
gdb /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ovStoreBuild
set args -O /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.ovlStore.BUILDING -G /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/ecoli.gkpStore -M 2-8 -L /Users/kevinmckernan/canu-1.4/Darwin-amd64/bin/ecoli-auto/correction/1-overlapper/ovljob.files
bt
```

That will give the trace of what the program was doing when it crashed.

**Comment by KevinMcKernan** [comment]
Looks like El Capitan doesn't have GDB and installing it requires something else. Going down this rabbit hole next.
http://stackoverflow.com/questions/33162757/how-to-install-gdb-debugger-in-mac-osx-el-capitan

I did get Canu to install on a linux AWS box and run in no time. Mac OS X makes everything worse.

**Comment by skoren** [comment]
Ah yes that's right, I forgot OS X makes GDB installation a pain. Don't worry about that for now, you can test again after we find out what's going on with the other ovlStore issue.

**Comment by KevinMcKernan** [comment]
I have a new Mac Pro coming in tomorrow with 64Gb RAM. This one is 5 years old with a patch work of upgrades. Let me try a fresh install on that one on Friday and see if that clears things up.

**Comment by brianwalenz** [comment]
This should be fixed now.  Open a new issue if needed - this one has drifted significantly.

For gdb, I've had good luck with the slightly older 'gdb-apple' in MacPorts.

**Comment by KevinMcKernan** [comment]
Thank you for the help!
We are porting it over to unbuntu.
Just got a new Mac that I will test out as well.
Does Canu merge ILMN data with MinION

**Comment by brianwalenz** [comment]
No Illumina support at all.

**Comment by drashutosh** [comment]
Hi 
I am trying to make an assembly of nanopore data with canu v1.7 , how while running my script I am finding the following error 
 
![image](https://user-images.githubusercontent.com/25049809/37447758-96c24208-2877-11e8-8add-1347bd00c7d8.png)

I am looking for suggestions regarding this error

**Comment by skoren** [comment]
Please don't use old issues, open a new one. 

The error message indicates no input was given to Canu. What's your command to run Canu and are you running this on a new empty folder or continuing an existing run.
